content/docs/user-guide/testing/getting-started/_index.md

```diff
@@ -0,0 +1,233 @@
+
+When building production C++ code for O3DE features, that code gets built into a library to later load into the engine. When writing C++ tests for this code, the test-code should be built into a separate library with a dependency on the production library. Building separate test libraries has the advantage of keeping production binaries lean and easy to ship, free of extraneous ```#if defined``` blocks for test code, and ensures we test the exact same interfaces that we ship.
+
+### Registering new C++ tests
+
+While reading the upcoming steps, refer to the following `CMakeLists.txt` example:
+
+```
+# Preexisting module registration from Step 0
+ly_add_target(
+    NAME MyModuleName
+    NAMESPACE Gem
+    FILES_CMAKE
+        mymodule_files.cmake
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            Source
+        PUBLIC
+            Include
+    BUILD_DEPENDENCIES
+        PRIVATE
+            Legacy::CryCommon
+    COMPONENT
+        MyModuleComponent
+)
+
+# New test module registration from Step 1
+# this configures CMake to also build a library called MyModuleName.Tests.so (.dll on Windows)
+# the rest of CMake will refer to this freshly created module by its NAME ("MyModuleName.Tests")
+if(PAL_TRAIT_BUILD_SUPPORTS_TESTS)  # only create these test modules if the target platform supports testing
+
+    ly_add_target(
+        NAME MyModuleName.Tests SHARED
+        NAMESPACE Gem
+        FILES_CMAKE
+            mymodule_test_files.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                Tests
+                Source
+        BUILD_DEPENDENCIES
+            PRIVATE
+                AZ::AzTest
+                Legacy::CryCommon
+                Gem::MyModuleName
+        COMPONENT
+            MyModuleComponent
+    )
+
+    # Register new test module from Step 2
+    ly_add_googletest(
+        NAME MyModuleName.Tests
+    )
+
+endif()
+```
+
+#### Step 0: Add a production build target
+
+Before configuring tests, you should first define the library which gets tested. This is defined in a `CMakeLists.txt`, which typically lives in the same directory as the code or in a parent directory. For more information on configuring `CMakeLists.txt`, please read more about adding library targets at [Getting Started with the CMake Build System](/docs/user-guide/build/). Note that you only need to define a library that will be built; you can start configuring and writing tests before you finish writing the production code.
```

Comment:
```suggestion
Before configuring tests, you must first define the library that you want to test. This is defined in a `CMakeLists.txt`, which lives in the same directory as the code or in a parent directory. For more information on configuring `CMakeLists.txt`, please read more about adding library targets at [Getting Started with the CMake Build System](/docs/user-guide/build/). Note that you only need to define the library that you want to build. You can start configuring and writing tests before you finish writing the production code.
```
- Use "must" for required actions, and "should" for recommendations or probabilities. 
- Changed to "that you want to test" and "that you want to build." to clarify that the developer is the one choosing which library to test and build.

Reply:
Regarding "should"/"must" usage - we try and write to [IETF 2119](https://datatracker.ietf.org/doc/html/rfc2119) as this is what programmers interpret these words as (consciously or not - they've been reading them with these meanings their whole career.)

Reply:
The ordering a grey-area to me, but I think the recommendation is strong enough to default to a must.